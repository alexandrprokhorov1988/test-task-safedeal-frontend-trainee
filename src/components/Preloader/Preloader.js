import React from 'react';
import './Preloader.css';

function Preloader() {

  return (
    <div className="preloader">
      <div className="preloader__img"/>
      <p className="preloader__text">Идет поиск фотографий...</p>
    </div>
  );
}

export default Preloader;
