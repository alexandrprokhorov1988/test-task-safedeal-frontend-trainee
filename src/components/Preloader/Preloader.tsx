import React from 'react';
import './Preloader.css';

/**
 * Preloader component.
 */
const Preloader = () => {

  return (
    <div className="preloader">
      <div className="preloader__img"/>
      <p className="preloader__text">Идет поиск фотографий...</p>
    </div>
  );
};

export default Preloader;
