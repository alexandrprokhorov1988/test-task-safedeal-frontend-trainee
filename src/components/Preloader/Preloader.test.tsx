import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';

import Preloader from './Preloader';

let container: any = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Preloader component', () => {
  it('Рендер Preloader с текстом', () => {
    act(() => {
      render(<Preloader />, container);
    });
    expect(container.textContent).toBe('Идет поиск фотографий...');
  });

  it('Preloader component', () => {
    const result = renderer
      .create(<Preloader />)
      .toJSON();
    expect(result).toMatchSnapshot();
  });
});
