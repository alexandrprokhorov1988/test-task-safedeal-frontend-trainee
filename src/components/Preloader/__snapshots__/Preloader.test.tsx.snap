// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader component Preloader component 1`] = `
<div
  className="preloader"
>
  <div
    className="img"
  />
  <p
    className="text"
  >
    Идет поиск фотографий...
  </p>
</div>
`;
